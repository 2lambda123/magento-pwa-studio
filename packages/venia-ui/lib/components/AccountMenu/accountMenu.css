.root {
    background-color: rgb(var(--venia-global-color-background));
    border-radius: 1px;
    box-shadow: 1px 1px 5px rgb(var(--venia-global-color-gray-darker));
    opacity: 0;
    position: fixed;
    right: -100vw;
    top: 5rem;
    transform: translate3d(0, -8px, 0);
    transition-duration: 192ms;
    transition-timing-function: var(--venia-global-anim-out);
    transition-property: opacity, transform, visibility;
    visibility: hidden;
    width: 27.5rem;
    min-height: 18rem;
}

.root_open {
    composes: root;

    opacity: 1;
    right: 2rem;
    padding: 0.5rem;
    transform: translate3d(0, 4px, 0);
    transition-duration: 224ms;
    transition-timing-function: var(--venia-global-anim-in);
    visibility: visible;
}

.createAccount {
    padding: 2rem;
}

.forgotPassword {
    padding: 2rem;
}

.loading {
    padding: 2rem;
}
