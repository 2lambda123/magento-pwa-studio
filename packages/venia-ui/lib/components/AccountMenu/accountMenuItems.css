.link {
    composes: root from '../LinkButton/linkButton.css';

    color: rgb(var(--venia-global-color-gray-700));
    justify-content: start;
    text-decoration: none;
}

.link:hover {
    color: rgb(var(--venia-global-color-gray-600));
}

.signOut {
    composes: link;

    border-top: 2px solid rgb(var(--venia-global-color-gray));
    padding-top: 1rem;
}
