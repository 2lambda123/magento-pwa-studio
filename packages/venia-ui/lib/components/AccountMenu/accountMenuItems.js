import React, { Fragment } from 'react';
import { func, shape, string } from 'prop-types';

import { Link } from '@magento/venia-drivers';
import { mergeClasses } from '@magento/venia-ui/lib/classify';

import defaultClasses from './accountMenuItems.css';

const MENU_ITEMS = [
    { name: 'Order History', url: '' },
    { name: 'Store Credit & Gift Cards', url: '' },
    { name: 'Favorites Lists', url: '' },
    { name: 'Address Book', url: '' },
    { name: 'Saved Payments', url: '' },
    { name: 'Communications', url: '/communications' },
    { name: 'Account Information', url: '' }
];

const AccountMenuItems = props => {
    const { handleSignOut, handleClose } = props;

    const classes = mergeClasses(defaultClasses, props.classes);

    const menuItems = MENU_ITEMS.map(item => {
        return (
            <Link
                className={classes.link}
                to={item.url}
                key={item.name}
                onClick={handleClose}
            >
                {item.name}
            </Link>
        );
    });

    return (
        <Fragment>
            {menuItems}
            <button
                className={classes.signOut}
                onClick={handleSignOut}
                type="button"
            >{`Sign Out`}</button>
        </Fragment>
    );
};

export default AccountMenuItems;

AccountMenuItems.propTypes = {
    classes: shape({
        link: string,
        signOut: string
    }),
    handleSignOut: func,
    handleClose: func
};
