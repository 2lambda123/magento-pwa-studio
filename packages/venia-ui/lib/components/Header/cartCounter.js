import React from 'react';
import { number, shape, string } from 'prop-types';

import { mergeClasses } from '../../classify';
import defaultClasses from './cartCounter.css';

const CartCounter = props => {
    const { numItems } = props;

    const classes = mergeClasses(defaultClasses, props.classes);

    // Don't show anything both when we don't have data to show, and when numItems is zero (0).
    if (!numItems) {
        return null;
    }

    return <span className={classes.root}>{numItems}</span>;
};

CartCounter.propTypes = {
    classes: shape({
        root: string
    }),
    numItems: number
};

export default CartCounter;
