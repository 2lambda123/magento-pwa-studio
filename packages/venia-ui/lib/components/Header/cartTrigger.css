.root {
    composes: root from '../clickable.css';
    height: 3rem;
    min-width: 3rem;
    position: relative;
    z-index: 1;
    display: flex;
    align-content: center;
    justify-content: center;
}

.counter {
    top: 0.375rem;
    position: absolute;
    left: 1.5rem;
    font-weight: 600;
    box-sizing: border-box;
    padding: 0 0.25rem;
    background-color: rgb(var(--venia-brand-color-1-700));
    border-radius: 100px;
    color: rgb(var(--venia-global-color-background));
    line-height: 1.25rem;
    min-width: 1.25rem;
    font-size: 0.8125rem;
}
