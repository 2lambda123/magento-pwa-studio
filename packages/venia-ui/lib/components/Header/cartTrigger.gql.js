import gql from 'graphql-tag';

import { CartTriggerFragment } from './cartTriggerFragments.gql';

export const GET_ITEM_COUNT_QUERY = gql`
    query getCartDetails($cartId: String!) {
        cart(cart_id: $cartId) @connection(key: "Cart") {
            id
            ...CartTriggerFragment
        }
    }
    ${CartTriggerFragment}
`;
