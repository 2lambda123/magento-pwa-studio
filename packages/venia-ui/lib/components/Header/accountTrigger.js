import React, { Fragment } from 'react';
import { shape, string } from 'prop-types';

import { useAccountTrigger } from '@magento/peregrine/lib/talons/Header/useAccountTrigger';
import { mergeClasses } from '@magento/venia-ui/lib/classify';

import SIGN_OUT_MUTATION from '../../queries/signOut.graphql';
import AccountChip from '../AccountChip';
import AccountMenu from '../AccountMenu';
import defaultClasses from './accountTrigger.css';

/**
 * The AccountTrigger component is the call to action in the site header
 * that toggles the AccountMenu dropdown.
 *
 * @param {Object} props
 * @param {Object} props.classes - CSS classes to override element styles.
 */
const AccountTrigger = props => {
    const talonProps = useAccountTrigger({
        mutations: { signOut: SIGN_OUT_MUTATION }
    });
    const {
        accountMenuIsOpen,
        accountMenuRef,
        accountMenuTriggerRef,
        handleSignOut,
        handleTriggerClick,
        isUserSignedIn
    } = talonProps;

    const classes = mergeClasses(defaultClasses, props.classes);
    const rootClassName = accountMenuIsOpen ? classes.root_open : classes.root;

    return (
        <Fragment>
            <div className={rootClassName} ref={accountMenuTriggerRef}>
                <button
                    aria-label={'Toggle My Account Menu'}
                    className={classes.trigger}
                    onClick={handleTriggerClick}
                >
                    <AccountChip
                        fallbackText={'Sign In'}
                        shouldIndicateLoading={true}
                    />
                </button>
            </div>
            <AccountMenu
                handleSignOut={handleSignOut}
                isOpen={accountMenuIsOpen}
                isUserSignedIn={isUserSignedIn}
                ref={accountMenuRef}
            />
        </Fragment>
    );
};

export default AccountTrigger;

AccountTrigger.propTypes = {
    classes: shape({
        root: string,
        root_open: string,
        trigger: string
    })
};
