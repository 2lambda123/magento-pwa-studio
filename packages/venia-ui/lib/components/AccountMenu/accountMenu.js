import React from 'react';
import { shape, string } from 'prop-types';

import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { useAccountMenu } from '@magento/peregrine/lib/talons/Header/useAccountMenu';

import SignIn from '../SignIn/signIn';
import AccountMenuItems from './accountMenuItems';

import SIGN_OUT_MUTATION from '../../queries/signOut.graphql';

import defaultClasses from './accountMenu.css';

const AccountMenu = React.forwardRef((props, ref) => {
    const { accountMenuIsOpen, setAccountMenuIsOpen } = props;
    const talonProps = useAccountMenu({
        mutations: { signOut: SIGN_OUT_MUTATION },
        accountMenuIsOpen,
        setAccountMenuIsOpen
    });
    const {
        view,
        username,
        handleSignOut,
        handleForgotPassword,
        handleCreateAccount,
        updateUsername
    } = talonProps;

    const classes = mergeClasses(defaultClasses, props.classes);
    const rootClass = accountMenuIsOpen ? classes.root_open : classes.root;

    let dropdownContents = null;

    switch (view) {
        case 'ACCOUNT': {
            dropdownContents = <AccountMenuItems onSignOut={handleSignOut} />;

            break;
        }
        case 'FORGOT_PASSWORD': {
            dropdownContents = (
                // username will be used by forgot password component
                <div
                    className={classes.forgotPassword}
                    initialValues={{ email: username }}
                >
                    To be handled in PWA-77
                </div>
            );

            break;
        }
        case 'CREATE_ACCOUNT': {
            dropdownContents = (
                // username will be used by create account component
                <div
                    className={classes.createAccount}
                    initialValues={{ email: username }}
                >
                    To be handled in PWA-804
                </div>
            );

            break;
        }
        case 'SIGNIN':
        default: {
            dropdownContents = (
                <SignIn
                    classes={{
                        modal_active: classes.loading
                    }}
                    setDefaultUsername={updateUsername}
                    showCreateAccount={handleCreateAccount}
                    showForgotPassword={handleForgotPassword}
                />
            );

            break;
        }
    }

    return (
        <aside className={rootClass} ref={ref}>
            {dropdownContents}
        </aside>
    );
});

export default AccountMenu;

AccountMenu.propTypes = {
    classes: shape({
        root: string,
        root_open: string,
        link: string
    })
};
