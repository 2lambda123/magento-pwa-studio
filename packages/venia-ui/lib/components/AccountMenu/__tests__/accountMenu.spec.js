import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import { useAccountMenu } from '@magento/peregrine/lib/talons/Header/useAccountMenu';

import AccountMenu from '../accountMenu';

jest.mock('../accountMenuItems', () => 'AccountMenuItems');
jest.mock('../../SignIn/signIn', () => 'SignIn Component');

jest.mock('@magento/peregrine/lib/talons/Header/useAccountMenu', () => ({
    useAccountMenu: jest.fn().mockReturnValue({
        view: 'ACCOUNT',
        username: 'gooseton',
        handleSignOut: jest.fn(),
        handleForgotPassword: jest.fn(),
        handleCreateAccount: jest.fn(),
        updateUsername: jest.fn()
    })
}));

const defaultTalonProps = {
    view: 'ACCOUNT',
    username: 'gooseton',
    handleSignOut: jest.fn(),
    handleForgotPassword: jest.fn(),
    handleCreateAccount: jest.fn(),
    updateUsername: jest.fn()
};

const defaultProps = {
    accountMenuIsOpen: false,
    setAccountMenuIsOpen: jest.fn(),
    classes: {
        modal_active: 'modal_active_class'
    }
};

test('it renders AccountMenuItems when the user is signed in', () => {
    useAccountMenu.mockReturnValueOnce({
        ...defaultTalonProps,
        view: 'ACCOUNT'
    });

    // Act.
    const instance = createTestInstance(<AccountMenu {...defaultProps} />);

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});

test('it renders SignIn component when the view is SIGNIN', () => {
    useAccountMenu.mockReturnValueOnce({
        ...defaultTalonProps,
        view: 'SIGNIN'
    });

    // Act.
    const instance = createTestInstance(<AccountMenu {...defaultProps} />);

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});

test('it renders forgot password component when the view is FORGOT_PASSWORD', () => {
    useAccountMenu.mockReturnValueOnce({
        ...defaultTalonProps,
        view: 'FORGOT_PASSWORD'
    });

    // Act.
    const instance = createTestInstance(<AccountMenu {...defaultProps} />);

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});

test('it renders forgot password component when the view is CREATE_ACCOUNT', () => {
    useAccountMenu.mockReturnValueOnce({
        ...defaultTalonProps,
        view: 'CREATE_ACCOUNT'
    });

    // Act.
    const instance = createTestInstance(<AccountMenu {...defaultProps} />);

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});
