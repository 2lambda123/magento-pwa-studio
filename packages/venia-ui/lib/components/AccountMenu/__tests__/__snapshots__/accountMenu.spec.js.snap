// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders AccountMenuItems when the user is signed in 1`] = `
<aside>
  <AccountMenuItems
    onSignOut={[MockFunction]}
  />
</aside>
`;

exports[`it renders SignIn component when the view is SIGNIN 1`] = `
<aside>
  <SignIn Component
    classes={
      Object {
        "modal_active": undefined,
      }
    }
    setDefaultUsername={[MockFunction]}
    showCreateAccount={[MockFunction]}
    showForgotPassword={[MockFunction]}
  />
</aside>
`;

exports[`it renders forgot password component when the view is CREATE_ACCOUNT 1`] = `
<aside>
  <div
    initialValues={
      Object {
        "email": "gooseton",
      }
    }
  >
    To be handled in PWA-804
  </div>
</aside>
`;

exports[`it renders forgot password component when the view is FORGOT_PASSWORD 1`] = `
<aside>
  <div
    initialValues={
      Object {
        "email": "gooseton",
      }
    }
  >
    To be handled in PWA-77
  </div>
</aside>
`;
